README.md

```diff
@@ -16,6 +16,8 @@ Just drop this line in your Gemfile:
 gem 'safe-pg-migrations'
 ```
 
+**Note: Do not run migrations via PgBouncer connection. You must run migrations via a direct Postgres connection**
```

Comment:
In the description of the PR, you wrote that the transactional pooling mode of PgBouncer is the issue. I guess PgBouncer could be used in another mode without issue with safe-pg-migrations?

Would you mind updating your PR to include that point? I would be interesting to include a link to the ticket you created, it can be useful if someone seeks for more information.

Reply:
Yes, PgBoucner [can be configured to run in](https://www.pgbouncer.org/features.html) Statement pooling mode, which is even more aggressive than the Transactional mode, and Session mode.

Session mode is more like a pool. However, it still has the downside of littering the connection settings (e.g. `statement_timeout`).

So I'd prefer to keep this as a hard rule and not run via PgBouncer in either of its modes.

Reply:
I'm not sure that I get why session pooling mode is problematic with safe-pg-migrations?

Reply:
If something goes wrong, and we happen not to reset the settings made, they remain in the connection that goes to the pool.

> When a client connects, a server connection will be assigned to it for the whole duration it stays connected. When the client disconnects, the server connection will be put back into **pool**.

I was under the assumption that `safe-pg-migrations` can leave a setting hanging, but looking at [this](https://github.com/doctolib/safe-pg-migrations/blob/a13b8740fc1c3a2ae80d1abeb82df50f529d02d1/lib/safe-pg-migrations/plugins/statement_insurer.rb#L106), it seems that it correctly resets the setting on exception.

So session pooling mode should be fine. Let me add a correction.

Reply:
Great thanks :)

Reply:
Done.

Reply:
That's great, thanks!